config/config-server/src/com/thoughtworks/go/config/parts/XmlPartialConfigProvider.java

```diff
@@ -0,0 +1,126 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END***********************************/
+package com.thoughtworks.go.config.parts;
+
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.config.remote.PartialConfig;
+import com.thoughtworks.go.domain.WildcardScanner;
+import com.thoughtworks.go.domain.config.Configuration;
+import com.thoughtworks.go.domain.config.ConfigurationProperty;
+import com.thoughtworks.go.util.FileUtil;
+import org.jdom.input.JDOMParseException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class XmlPartialConfigProvider implements PartialConfigProvider {
+
+    private final String defaultPatter = "**/*.gocd.xml";
+
+    private MagicalGoConfigXmlLoader loader;
+
+    public  XmlPartialConfigProvider(MagicalGoConfigXmlLoader loader)
+    {
+        this.loader = loader;
+    }
+
+    @Override
+    public PartialConfig load(File configRepoCheckoutDirectory, PartialConfigLoadContext context) {
+        File[] allFiles = getFiles(configRepoCheckoutDirectory, context);
+
+        // if context had changed files list then we could parse only new content
+
```

Comment:
Please update method names to conform to java naming convention for - ParseFiles, CollectFragments, ParseFile